File: apps/android/meetingkit/src/main/java/ai/q9labs/chalk/meetingkit/ChalkMeetingState.kt
package ai.q9labs.chalk.meetingkit

data class ChalkMeetingState(
	val connection: String = "disconnected",
	val participants: List<ChalkParticipant> = emptyList(),
	val lastError: String? = null,
)

data class ChalkParticipant(
	val id: String,
	val displayName: String,
	val audioEnabled: Boolean = false,
	val videoEnabled: Boolean = false,
	val role: String? = null,
)

File: apps/ios/ChalkMeetingKit/Sources/ChalkMeetingKit/ChalkMeetingState.swift
import Foundation

public struct ChalkMeetingState: Equatable {
	public var connection = "disconnected"
	public var participants: [ChalkParticipant] = []
	public var lastError: String?

	public init() {}
}

public struct ChalkParticipant: Codable, Equatable, Identifiable {
	public let id: String
	public var displayName: String
	public var audioEnabled: Bool
	public var videoEnabled: Bool
	public var role: String?

	public init(
		id: String,
		displayName: String,
		audioEnabled: Bool = false,
		videoEnabled: Bool = false,
		role: String? = nil
	) {
		self.id = id
		self.displayName = displayName
		self.audioEnabled = audioEnabled
		self.videoEnabled = videoEnabled
		self.role = role
	}
}

